pkg/playback/http_recorder.go

```diff
@@ -0,0 +1,143 @@
+package playback
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+// A recordServer proxies requests to a remote host, and records all interactions.
+type recordServer struct {
+	recorder   *interactionRecorder
+	remoteURL  string
+	webhookURL string
+}
+
+func newRecordServer(remoteURL string, webhookURL string) (httpRecorder recordServer) {
+	httpRecorder = recordServer{}
+	httpRecorder.remoteURL = remoteURL
+	httpRecorder.webhookURL = webhookURL
+
+	return httpRecorder
+}
+
+func (httpRecorder *recordServer) insertCassette(writer io.Writer) error {
+	recorder, err := newInteractionRecorder(writer, httpRequestToBytes, httpResponseToBytes)
+	if err != nil {
+		return err
+	}
+	httpRecorder.recorder = recorder
+
+	return nil
+}
+
+func (httpRecorder *recordServer) webhookHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
can you add docstrings for `webhookHandler` and `handler`? Just a little reminder of what the functions are for/how they are called could go a long way

Reply:
Done! 85caf865a9d8b828a9635da735339795656f2173